src/main/java/com/greenfox/poker/service/GameService.java

```diff
@@ -1,82 +1,150 @@
+    if (!gameStateMap.isEmpty()) {
+      if (!gameStateMap.get(gameId).getPlayers().isEmpty()) {
+        for (GamePlayer player : gameStateMap.get(gameId).getPlayers()) {
+          if (player.getId() == userId) {
+            return true;
+          }
+        }
+      }
     }
-    return new ResponseEntity(new StatusError("fail", "game id doesn’t exist"),
-            HttpStatus.NOT_FOUND);
+    return false;
+  }
+
+  public void createNewPlayerAndAddToGame(long playerId, long gameId, long chipsToPlayWith){
+    PokerUserDTO player = dtoService.userDTOHashMap.get(playerId);
+    GamePlayer newPlayer = new GamePlayer(chipsToPlayWith, player);
+    gameStateMap.get(gameId).getPlayers().add(newPlayer);
+  }
+
+  public ResponseType joinPlayerToGame(long playerId, long gameId, long chipsToPlayWith){
+    String gameName = gameRepo.findOne(gameId).getName();
+    String playerName = dtoService.userDTOHashMap.get(playerId).getUsername();
+    dtoService.deductChipsFromAvailableChips(chipsToPlayWith, playerId);
+    createNewPlayerAndAddToGame(playerId,gameId,chipsToPlayWith);
+    return new StatusError("success", playerName + " joined game: " + gameName);
+  }
+
+  public int getTableBigBlind (long gameId){
+    return gameRepo.findOne(gameId).getBigBlind();
   }
 
-  public ResponseEntity<?> saveNewGame(Game game, BindingResult bindingResult) {
-    if (bindingResult.hasErrors()) {
-      return new ResponseEntity(errorMessageService.respondToMissingParameters(bindingResult),
-              HttpStatus.BAD_REQUEST);
-    } else {
-      return new ResponseEntity(saveGame(game), HttpStatus.OK);
+  public ResponseType leaveGame(long gameId, long playerId){
+    List<GamePlayer> playersAtTable = getGameStateById(gameId).getPlayers();
+    for (GamePlayer player : playersAtTable){
+      if (playerId == player.getId()){
+        playersAtTable.remove(player);
+      }
     }
+    getGameStateById(gameId).setPlayers(playersAtTable);
+    String playerName = dtoService.userDTOHashMap.get(playerId).getUsername();
+    String gameName = gameRepo.findOne(gameId).getName();
+    return new StatusError("success", playerName + " left game: " + gameName);
   }
 
-  public ResponseEntity<?> joinTable(long id, ChipsToJoinGame chips, String token) {
-//    TokenService tokenService = new TokenService();
-//    String username = tokenService.getUsernameFromToken(token);
-    Game game = getGameById(id);
-    long gameStateId = game.getGamestateId();
-    return new ResponseEntity(getGameState(gameStateId), HttpStatus.OK);
+  public Integer getPlayerIndexFromGameState(long playerId, long gameId) {
+    List<GamePlayer> playersAtTable = getGameStateById(gameId).getPlayers();
+    Integer index = 0;
+    for (GamePlayer player : playersAtTable) {
+      if (playerId == player.getId()) {
+        index = playersAtTable.indexOf(player);
+      }
+    }
+    return index;
+  }
 
+  public void updateGame(long playerId, long gameId, PlayerAction playerAction){
```

Comment:
This is a function where you do a lot of different things.
Separate the contents below to separate, well described functions that do just one thing.